test/spec_server.rb

```diff
@@ -349,8 +371,9 @@ def start(*) [self.class, :started] end
       daemonize: false,
       server: 'webrick'
     )
-    t = Thread.new { server.start { |s| Thread.current[:server] = s } }
-    t.join(0.01) until t[:server] && t[:server].status != :Stop
+
+    t = start_server(server)
+
     body = if URI.respond_to?(:open)
```

Comment:
Why not have all that logic in the `server.start` block?
Is the server stopped when the block ends?

Reply:
I have no idea about the deeper implementation, I just tried to move the existing shared logic into a method.